scalameta/contrib/src/main/scala/scala/meta/contrib/implicits/Equality.scala

```diff
@@ -0,0 +1,20 @@
+package scala.meta.contrib.implicits
+
+import scala.language.higherKinds
+import scala.meta.Tree
+import scala.meta.contrib.equality.Equal
+
+trait Equality {
+
+  type Structurally[A <: Tree] = scala.meta.contrib.equality.Structurally[A]
```

Comment:
I kind of regret adding the `<: Tree` restriction on the equality type classes, it propagates to everywhere for a questionable benefit, I opened #706. I can fix it later.